### ISSUE TITLE
Dynamic Data Retrieval

### ISSUE THREAD

**Issue by jmadeano** [babe-project/IteratedNarration#1]
It looks like the dynamic data-retrieval queries the database looking for all valid chain ends and makes a new chain for good measure, and then selects from those options. However, it seems as though there is a window of time between a participant being assigned a chain and them finishing the experiment where it is possible for another participant to be assigned the same generation of the same chain. I see that this experiment hasn't been updated in a while, so I was wondering if this issue might have been discovered and fixed in other repos? Should there/can there be an initial post to the database as soon as a chain is selected?

**Comment by elisakreiss** [comment]
Yes, that is correct. We don't have a clean solution for this yet, but plan to incorporate this in the future. 
When you have students/participants do the task at home over a longer time frame, it is very unlikely that this problem will arise, because they would need to get the same story from the same chain at the same time. (I'm currently running a slightly modified experiment with five stories that can appear in either of two conditions, five chains per condition with five generations each. Participants are students that can participate at some point in the time course of a month and we haven't lost a single data point due to this.) In an online study environment where a lot of participants work on your study at the same time (e.g., on MTurk), you should make sure to initialize enough chains so that the probability for a double submission decreases. It is still quite likely that you will have some double submissions. The template detects these immediately and automatically terminates the forked chain of the last submission. In that sense you're still keeping the damage fairly small. In my studies the amount of double submissions was not very high and therefore not problematic for me so far.
Nevertheless, we know that this is not yet the cleanest solution, and it remains something that we want to change in the long-term.

**Comment by x-ji** [comment]
This might be helped by the interactive experiment module which is being developed. This module allows communication between all participants who are simultaneously online. It's intended for experiments which require interactions between the participants, but might also help solve this issue (e.g. detects that more than 1 participants have simultaneously accepted the task and perform some special work on the second participant's experiment)
